src/main/java/sirius/db/jdbc/SmartQuery.java

```diff
@@ -254,41 +247,15 @@ public boolean exists() {
         return copy().fields(SQLEntity.ID).first().isPresent();
     }
 
-    /**
-     * Deletes all matches using the {@link OMA#delete(BaseEntity)}.
-     * <p>
-     * Note that for very large result sets, we perform a blockwise strategy. We therefore iterate over
-     * the results until the timeout ({@link #queryIterateTimeout} is reached). In this case, we abort the
-     * iteration, execute the query again and continue deleting until all entities are gone.
-     *
-     * @param entityCallback a callback to be invoked for each entity to be deleted
-     */
     @Override
     public void delete(@Nullable Consumer<E> entityCallback) {
-        if (forceFail) {
```

Comment:
dont we need `foreFail` anymore?

Reply:
streamblockwise does it internally, but it only does at block level. Therefore its wise to check at record level, since the cascade can propagate to child entities.